dozer-sink-aerospike/src/lib.rs

```diff
@@ -0,0 +1,960 @@
+                    let mut r = AsRecord(record.assume_init_mut());
+                    self.client.insert(k.as_ptr(), r.as_mut_ptr())?;
+                }
+            }
+            Operation::Delete { old } => {
+                let mut key = MaybeUninit::uninit();
+                unsafe {
+                    self.init_key(key.as_mut_ptr(), &old, &mut allocated_strings)?;
+                    let k = Key(key.assume_init_mut());
+                    self.client.delete(k.as_ptr())?;
+                }
+            }
+            Operation::Update { old, new } => {
+                let mut key = MaybeUninit::uninit();
+                let mut record = MaybeUninit::uninit();
+                unsafe {
+                    self.init_key(key.as_mut_ptr(), &old, &mut allocated_strings)?;
+                    let k = Key(key.assume_init_mut());
+                    self.init_record(record.as_mut_ptr(), &new, &mut allocated_strings)?;
+                    let mut r = AsRecord(record.assume_init_mut());
+                    self.client.update(k.as_ptr(), r.as_mut_ptr())?;
+                }
+            }
+            Operation::BatchInsert { new } => {
+                // Create an as_batch_write_record for each key
+                // Create an as_operations for each bin and assign them to the
+                // as_batch_write_record
+                let mut batch = unsafe {
+                    let mut batch = MaybeUninit::uninit();
+                    as_batch_records_init(batch.as_mut_ptr(), new.len() as u32);
+                    Batch(batch.assume_init())
+                };
+                // Wrapper type here, so `as_operations_destroy` is called, even
+                // when an error occurs
+                let mut operations = Operations::new(new.len());
+                for dozer_record in new.iter() {
+                    unsafe {
+                        let record = as_batch_write_reserve(batch.as_ptr());
+                        let ops = operations.next(dozer_record.values.len());
+                        if ops.is_null() {
+                            return Err(AerospikeSinkError::CreateRecordError);
+                        }
+                        self.init_ops(ops, dozer_record, &mut allocated_strings)?;
+                        (*record).ops = ops;
+                        self.init_key(
+                            &mut (*record).key as *mut as_key,
+                            dozer_record,
+                            &mut allocated_strings,
+                        )?;
+                    }
+                }
+                unsafe {
+                    self.client.write_batch(batch.as_ptr())?;
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+struct Operations(Vec<MaybeUninit<as_operations>>);
+impl Operations {
+    fn new(size: usize) -> Self {
+        Self(Vec::with_capacity(size))
+    }
+
+    fn next(&mut self, n_bins: usize) -> *mut as_operations {
```

Comment:
I assume we cannot move `as_operations`, so this function is `unsafe` because it should only be called at most `size` times.